iree/base/internal/status_builder.h

```diff
@@ -91,79 +103,110 @@ class ABSL_MUST_USE_RESULT StatusBuilder {
-    SourceLocation location IREE_LOC_CURRENT_DEFAULT_ARG);
-StatusBuilder DeadlineExceededErrorBuilder(
-    SourceLocation location IREE_LOC_CURRENT_DEFAULT_ARG);
-StatusBuilder FailedPreconditionErrorBuilder(
-    SourceLocation location IREE_LOC_CURRENT_DEFAULT_ARG);
-StatusBuilder InternalErrorBuilder(
-    SourceLocation location IREE_LOC_CURRENT_DEFAULT_ARG);
-StatusBuilder InvalidArgumentErrorBuilder(
-    SourceLocation location IREE_LOC_CURRENT_DEFAULT_ARG);
-StatusBuilder NotFoundErrorBuilder(
-    SourceLocation location IREE_LOC_CURRENT_DEFAULT_ARG);
-StatusBuilder OutOfRangeErrorBuilder(
-    SourceLocation location IREE_LOC_CURRENT_DEFAULT_ARG);
-StatusBuilder PermissionDeniedErrorBuilder(
-    SourceLocation location IREE_LOC_CURRENT_DEFAULT_ARG);
-StatusBuilder UnauthenticatedErrorBuilder(
-    SourceLocation location IREE_LOC_CURRENT_DEFAULT_ARG);
-StatusBuilder ResourceExhaustedErrorBuilder(
-    SourceLocation location IREE_LOC_CURRENT_DEFAULT_ARG);
-StatusBuilder UnavailableErrorBuilder(
-    SourceLocation location IREE_LOC_CURRENT_DEFAULT_ARG);
-StatusBuilder UnimplementedErrorBuilder(
-    SourceLocation location IREE_LOC_CURRENT_DEFAULT_ARG);
-StatusBuilder UnknownErrorBuilder(
-    SourceLocation location IREE_LOC_CURRENT_DEFAULT_ARG);
+StatusBuilder AbortedErrorBuilder(SourceLocation location);
+StatusBuilder AlreadyExistsErrorBuilder(SourceLocation location);
+StatusBuilder CancelledErrorBuilder(SourceLocation location);
+StatusBuilder DataLossErrorBuilder(SourceLocation location);
+StatusBuilder DeadlineExceededErrorBuilder(SourceLocation location);
+StatusBuilder FailedPreconditionErrorBuilder(SourceLocation location);
+StatusBuilder InternalErrorBuilder(SourceLocation location);
+StatusBuilder InvalidArgumentErrorBuilder(SourceLocation location);
+StatusBuilder NotFoundErrorBuilder(SourceLocation location);
+StatusBuilder OutOfRangeErrorBuilder(SourceLocation location);
+StatusBuilder PermissionDeniedErrorBuilder(SourceLocation location);
+StatusBuilder UnauthenticatedErrorBuilder(SourceLocation location);
+StatusBuilder ResourceExhaustedErrorBuilder(SourceLocation location);
+StatusBuilder UnavailableErrorBuilder(SourceLocation location);
+StatusBuilder UnimplementedErrorBuilder(SourceLocation location);
+StatusBuilder UnknownErrorBuilder(SourceLocation location);
+
+// Returns a StatusBuilder using a status code derived from |errno|.
+StatusBuilder ErrnoToCanonicalStatusBuilder(int error_number,
+                                            SourceLocation location);
+
+#if defined(IREE_PLATFORM_WINDOWS)
+
+// Returns a StatusBuilder with a status describing the |error| and |location|.
+StatusBuilder Win32ErrorToCanonicalStatusBuilder(uint32_t error,
+                                                 SourceLocation location);
+
+#endif  // IREE_PLATFORM_WINDOWS
 
 }  // namespace iree
 
+// Override the C macro with our C++ one.
+// For files that only include the C API header they'll only support
```

Comment:
Ooo, I like how this supports both C and C++ seamlessly.

Reply:
It took some fiddling, but not needing to worry about which language you're in or which macro you should use is really nice :)
I already don't want to write any more code in the old way.